### ISSUE TITLE
Unable to recognize the different changes of characters

### ISSUE THREAD

**Issue by aaaaalun** [wangle53/TransCD#22]
Hello, author. I used your model and it could detect changes from nothing to something, but now I can't detect the differences in characters. I have no idea why. I trained on transCD and AnoDFDNet, both with a loss of 0.009, which is very small. However, the f1 values tested were all 0. All my data were self-synthesized, as shown in the figure：

<img width="535" height="245" alt="Image" src="https://github.com/user-attachments/assets/d47b803d-23aa-4eaa-a784-fba01e466af9" />

t0：
<img width="278" height="103" alt="Image" src="https://github.com/user-attachments/assets/aa6b7165-25da-47df-bb4e-ed148bd0a278" />

t1：
<img width="635" height="225" alt="Image" src="https://github.com/user-attachments/assets/3c20efe1-b07a-46e3-b6b8-63822a67d14f" />

gt：
<img width="429" height="195" alt="Image" src="https://github.com/user-attachments/assets/7df9448e-4cb6-4aa5-8dc8-2c86d78f12ff" />

These graphs are all just some of my screenshots. Is there anything wrong with such data?

**Comment by wangle53** [comment]
I’m not sure what went wrong with your data. However, based on the information you have provided so far, the model does not appear to be trained effectively. You may want to investigate the issue from the following aspects:
1 Whether the ground-truth (GT) values are correct when computing the loss, and whether they are binary (0 and 1);
2 Whether the learning rate is set properly—trying different learning rates may help;
3 Whether the size of your dataset is too small.

**Comment by aaaaalun** [comment]
1. I previously set it to 255, but now I have changed it to 1.
2. I adjusted different learning rates, some large and some small.
3. I have provided 13,000 images.
I tried everything, but the result was still the same.
This is the image I synthesized myself. The mask is the character mask of w in that string of characters：
<img width="635" height="477" alt="Image" src="https://github.com/user-attachments/assets/1cc07870-51b3-43d6-9916-55a94618f319" />
<img width="631" height="472" alt="Image" src="https://github.com/user-attachments/assets/f7e9fe7a-194f-4aba-a4ba-d5532247db98" />

I saw an image in your paper that made me puzzled. This house is quite different. Why wasn't it detected as a change? Is it because when training image pairs, the house did not provide a mask with the value 1, so does the network understand such a change as unchanging? Does that mean the net has already regarded the house as an unimportant change? If only the road surface is focused on, will the subsequent provision of two pictures containing changes similar to houses also be understood by the net as unimportant changes?
Just as in the picture above, I generated defect data on the picture containing the QR code. Then, is there any change containing the QR code that cannot be detected later?
<img width="760" height="259" alt="Image" src="https://github.com/user-attachments/assets/9f8d94c3-5db3-4fa8-a2bf-e50e0539febf" />
